components/collective-navbar/ActionsMenu.js

```diff
@@ -132,11 +152,92 @@ export const getContributeRoute = collective => {
   return { route, params };
 };
 
-const CollectiveNavbarActionsMenu = ({ collective, callsToAction, hiddenActionForNonMobile }) => {
+const getSecondCTA = (enabledCTAs, hasRequestGrant, mainAction) => {
+  if (!hasRequestGrant) {
+    return enabledCTAs;
+  } else {
+    return difference(enabledCTAs, mainAction);
+  }
+};
+
+const getSecondActionButton = (callToAction, collective) => {
```

Comment:
:ice_cream: We're duplicating a lot of code for generating the links here. Would be great if we could find a way to avoid that in the future.

Reply:
Could we address this now if it makes things more maintainable for the future?

Reply:
If we go with a refactor, I also think that this component is not the right place to implement this button. I would expect the secondary action to be rendered next to the primary action, so https://github.com/opencollective/opencollective-frontend/blob/emptyandspacing/components/collective-navbar/index.js#L580. Until now, `ActionsMenu`'s spec has been to render itself as a dropdown menu, rendering a button next to it seems a bit out of scope. But if it makes the implementation easier somehow we can live with it.